spatial/r2/triangle.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r2
+
+import "math"
+
+// Triangle represents a triangle in 2D space and
+// is composed by 3 vectors corresponding to the position
+// of each of the vertices.
+type Triangle [3]Vec
+
+// Degenerate returns true if triangle's vertices are collinear to within
+// a certain tolerance.
+func (t Triangle) Degenerate(tol float64) bool {
+	// TODO(soypat): Must be a better way to do this.
+	// https://stackoverflow.com/questions/33037449/given-three-side-of-a-triangle-how-can-i-define-whether-it-is-a-degenerate-tria
+	a, b, c := t.orderedLengths()
+	// The sum of the shorter sides of a valid triangle is always
+	// larger than the longest of the sides.
+	return a+b < c+tol
+}
+
+// Area returns the surface area of the triangle.
+func (t Triangle) Area() float64 {
+	// William M. Kahan (24 March 2000). "Miscalculating Area and Angles of a Needle-like Triangle"
```

Comment:
Add a link, https://people.eecs.berkeley.edu/~wkahan/Triangle.pdf, and which formula was used. I can't find a doi for the paper.